### ISSUE TITLE
Database queue connection jobs run in app root directory path instead of public

### ISSUE THREAD

**Issue by KeithTurkowski** [laravel/framework#45679]
- Laravel Version: 9.47
- PHP Version: 8.0
- Database Driver & Version: MySQL

### Description:

When I run a job using `QUEUE_CONNECTION=database` it runs in a different directory path than when the job is run using `QUEUE_CONNECTION=sync`

Specifically, sync runs in ~/public/, and database runs in ~/

### Steps To Reproduce:

Create a new Laravel app with support for database queue connections.

Create a Job that executes `echo getcwd()` in the handle method.

In the `.env` set: `QUEUE_CONNECTION=sync`

Dispatch the job via a route and observe that the directory path displayed is in 'public', e.g. `/home/laravel_root/public` 

In the `.env` set: `QUEUE_CONNECTION=database`

Dispatch the job via a route, run `php artisan queue:work` and observe that the directory path displayed is in the root, e.g. `/home/laravel_root`

### Notes:
I believe this behavior changed with Laravel 6

I don't find anything related to this in Laravel upgrade notes, queue / jobs documentation, or in the config/queue.php

**Comment by timacdonald** [comment]
I don't know that this is a bug really.

When a HTTP request comes in, it is passed to `/public/index.php`. When a queue worker is run, it is run from `/` via `php artisan queue:work`.

They have different working directories as they are started from different working directories. Although at first it feels weird, I believe it makes sense and is the expected behaviour.

**Comment by KeithTurkowski** [comment]
Well, it's a bit confusing behavior, so I'd think at least it should be noted in the documentation.

It's also a bit strange, with the database queue connection, if I change directory path in the current job, and then dispatch a new job, the new job will run in whatever directory path the dispatching job was in.

I encountered this issue because I was using devfactory/imagecache to cache images, and it wasn't working properly when I tried to use a Job to cache the images.

For now I'm solving the problem by adding this to all Jobs...

`if(config('queue.default') == "database" && !str_ends_with(getcwd(), 'public')) { chdir('public'); }`

Maybe the behavior isn't a bug, but I'd suggest to note in the documentation, and perhaps add a setting in `config/queue.php` to specify the default starting path for each driver connection.
